Sources/TuistKit/Generator/ProjectGroups.swift

```diff
@@ -1,34 +1,59 @@
 import Basic
 import Foundation
+import TuistCore
 import xcodeproj
 
+enum ProjectGroupsError: FatalError, Equatable {
+    case missingGroup(String)
+
+    var description: String {
+        switch self {
+        case let .missingGroup(group):
+            return "Couldn't find group: \(group)"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .missingGroup:
+            return .bug
+        }
+    }
+}
+
 class ProjectGroups {
     // MARK: - Attributes
 
     let main: PBXGroup
     let products: PBXGroup
     let projectManifest: PBXGroup
-    let project: PBXGroup
     let frameworks: PBXGroup
     let playgrounds: PBXGroup?
-    let pbxproj: PBXProj
+
+    private let pbxproj: PBXProj
+    private let projectGroups: [String: PBXGroup]
+    private let knownGroups: OrderedSet<PBXGroup>
```

Comment:
I think the `knownGroups` is a bit ambiguous. What's known in the context of an Xcode project? If we can't come up with a better name I'd add a comment.

Reply:
👍 will rename it to `allPredefinedGroups`